sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/HttpTimeoutPolicy.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.http;
+
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.implementation.ResourceType;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.RxDocumentServiceResponse;
+import io.netty.handler.codec.http.HttpMethod;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class HttpTimeoutPolicy {
+    final public static HttpTimeoutPolicy getTimeoutPolicy(RxDocumentServiceRequest request) {
+
+        if (OperationType.QueryPlan.equals(request.getOperationType()) ||
+            request.isAddressRefresh() ||
+            request.getResourceType() == ResourceType.PartitionKeyRange) {
+            return HttpTimeoutPolicyControlPlaneHotPath.instance;
+        }
+
+        return HttpTimeoutPolicyDefault.instance;
+    }
+
+    public abstract Duration maximumRetryTimeLimit();
+
+    public abstract Integer totalRetryCount();
+
+    public abstract List<ResponseTimeoutAndDelays> getTimeoutList();
+
+    public abstract Boolean isSafeToRetry(HttpMethod httpMethod);
+
+    public abstract Boolean shouldRetryBasedOnResponse(HttpMethod requestHttpMethod, Mono<RxDocumentServiceResponse> responseMessage);
```

Comment:
Same here, check if we can use primitive types - they make the code easier to read and debug, as they can never be null. Reduces the different possible values. 
Also, this method is never used. See if we need it or not.

Reply:
Removed the method, we are not using it.